app/src/organisms/ConfigurePipette/ConfigForm.tsx

```diff
@@ -138,26 +142,38 @@ export class ConfigForm extends React.Component<ConfigFormProps> {
           // TODO(bc, 2021-05-18): this should probably be (parsed < min || parsed > max) so we're not accidentally comparing a string to a number
           (parsed < min || value > max)
         ) {
-          set(errors, name, `Min ${min} / Max ${max}`)
+          set(errors, name, {
+            type: 'numberError',
+            message: `Min ${min} / Max ${max}`,
+          })
         }
       }
     })
 
     const plungerGroupError =
       'Please ensure the following: \n top > bottom > blowout > droptip'
-    const top = this.getFieldValue('top', plungerFields, values)
-    const bottom = this.getFieldValue('bottom', plungerFields, values)
-    const blowout = this.getFieldValue('blowout', plungerFields, values)
-    const dropTip = this.getFieldValue('dropTip', plungerFields, values)
+    const top = getFieldValue('top', plungerFields, values)
+    const bottom = getFieldValue('bottom', plungerFields, values)
+    const blowout = getFieldValue('blowout', plungerFields, values)
+    const dropTip = getFieldValue('dropTip', plungerFields, values)
     if (top <= bottom || bottom <= blowout || blowout <= dropTip) {
-      set(errors, 'plungerError', plungerGroupError)
+      set(errors, 'plungerError', {
+        type: 'plungerError',
+        message: plungerGroupError,
+      })
     }
 
     return errors
   }
 
-  getInitialValues: () => FormValues = () => {
-    const fields = this.getVisibleFields()
+  const resolver: Resolver<FormValues> = values => {
+    let errors = {}
+    errors = validate(values, errors)
```

Comment:
probably we don't need to use `let` here, do we?

```suggestion
    const errors = validate(values, errors)
```